vllm/distributed/parallel_state.py

```diff
@@ -674,6 +678,37 @@ def model_parallel_is_initialized():
     return (_TP is not None and _PP is not None)
 
 
+TP_STATE_PATCHED = False
+
+
+@contextlib.contextmanager
+def patch_tensor_parallel_group(world_group: Optional[GroupCoordinator],
+                                tp_group: Optional[GroupCoordinator]):
+    """Patch the tp group temporarily until this function ends.
+    It requires the world group to be patched together to keep the integrity.
```

Comment:
The docstring should say something about the intended use -- for draft model in spec decode.